packages/shorebird_cli/lib/src/platform/apple.dart

```diff
@@ -26,7 +27,30 @@ enum ApplePlatform {
 
 /// A record containing the exit code and optionally link percentage
 /// returned by `runLinker`.
-typedef LinkResult = ({int exitCode, double? linkPercentage});
+@immutable
+class LinkResult {
+  /// Creates a new [LinkResult] representing failure.
+  const LinkResult.failure()
+    : _exitCodeObject = ExitCode.software,
+      linkPercentage = null,
+      linkMetadata = null;
+
+  /// Creates a new [LinkResult] representing success.
+  const LinkResult.success({required this.linkPercentage, this.linkMetadata})
+    : _exitCodeObject = ExitCode.success;
+
+  /// ExitCode.code isn't const, so store the actual object.
+  final ExitCode _exitCodeObject;
+
+  /// The exit code of the linker process.
+  int get exitCode => _exitCodeObject.code;
```

Comment:
why do we need to hold onto the object instead of just the underlying int?

Reply:
We don't.  The only advantage of this is to let us have const constructors.

Reply:
The reason why we can't have a const constructor if we hold onto the int directly is that ExitCode.code isn't const-compatible.

Reply:
Isn't the failure case's exit code always software? Can't we avoid passing an ExitCode altogether?

Reply:
Do you mean just store a bool instead of the code?